py-rattler/src/networking/client.rs

```diff
@@ -0,0 +1,42 @@
+use crate::networking::middleware::PyMiddleware;
+use pyo3::{pyclass, pymethods};
+use rattler_networking::{AuthenticationMiddleware, MirrorMiddleware};
+use reqwest_middleware::ClientWithMiddleware;
+
+#[pyclass]
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct PyClientWithMiddleware {
```

Comment:
not sure what is a better name, `PyClientWithMiddleware` or `PyClient`?

Reply:
Given the convention so far, I'd stick to `PyClientWithMiddleware`.